public/app/plugins/panel/piechart/PieChartPanel.tsx

```diff
@@ -77,16 +77,17 @@ export function PieChartPanel(props: Props) {
 
 function getLegend(props: Props, displayValues: FieldDisplay[]) {
   const legendOptions = props.options.legend ?? defaultLegendOptions;
+  const sortOrder = props.options.reduceOptions.sort ?? SortWithReducer.None;
 
   if (legendOptions.showLegend === false) {
     return undefined;
   }
   const total = displayValues.filter(filterDisplayItems).reduce(sumDisplayItemsReducer, 0);
 
   const legendItems: VizLegendItem[] = displayValues
-    // Since the pie chart is always sorted, let's sort the legend as well.
+    // Since the pie chart is always sorted, let's sort the legend as well if not sorted previously
     .sort((a, b) => {
-      if (isNaN(a.display.numeric)) {
+      if (sortOrder !== SortWithReducer.None || isNaN(a.display.numeric)) {
```

Comment:
Note: We have a related PR here to not apply this sort if the organize fields transformation is used https://github.com/grafana/grafana/pull/97713

Reply:
Perhaps we disable the option if that's the case?